src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -192,5 +223,26 @@ public static void EnsureArgumentsNotNull<TArgs>(TArgs args, Func<ILogger> logge
             _ = args ?? throw new ArgumentNullException(nameof(args));
             _ = logger ?? throw new ArgumentNullException(nameof(logger));
         }
+
+        /// <summary>
+        /// Throws an exception if the value passed in is not a valid config key.
+        /// </summary>
+        /// <exception cref="CommandException"></exception>
+        public static void ValidateConfigKey(string configKey)
+        {
+            if (!configKey.Equals(ConfigurationConstants.DependencyVersion, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.GlobalPackagesFolder, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.RepositoryPath, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.DefaultPushSource, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.HostKey, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.UserKey, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.PasswordKey, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.NoProxy, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.MaxHttpRequestsPerSource, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.SignatureValidationMode, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new CommandException(string.Format(CultureInfo.CurrentCulture, Strings.Error_ConfigSetInvalidKey, configKey));
+            }
```

Comment:
A potential issue with this approach is that as new entries are added to the `NuGet.Config` file, it requires update to be made to this logic in order for the command to remain functional.

Reply:
In [the PR for the Set command](https://github.com/NuGet/NuGet.Client/pull/5150#discussion_r1192428485), Andy suggested:

> A new public API in ConfigurationConstants which contains an IReadOnlyList<string> of all the config options, has a better chance of being maintained correctly when new config keys are added.

I'll bring these changes into this branch as well.